slybot/slybot/starturls/__init__.py

```diff
@@ -1,6 +1,101 @@
-from .generator import UrlGenerator
+from collections import OrderedDict as ODict
+from itertools import chain, product
 
+from scrapy.utils.spider import arg_to_iter
 
-class StartUrls():
-    def __call__(self, spec):
-        return spec
+import six
+from six.moves.urllib_parse import urlparse
+
+from .fragment_generator import FragmentGenerator
+from .generator import IdentityGenerator, UrlGenerator
+
+
+class StartUrlCollection(object):
+    def __init__(self, start_urls, generators=None, generator_type='start_urls'):
+        self.generators = generators
+        self.generator_type = generator_type
+        self.start_urls = map(self._url_type, start_urls)
+
+    def __iter__(self):
+        generated = (self._generate_urls(url) for url in self.start_urls)
+        for url in chain(*(arg_to_iter(g) for g in generated)):
+            yield url
+
+    def uniq(self):
+        return list(ODict([(s.key, s.spec) for s in self.start_urls]).values())
+
+    @property
+    def allowed_domains(self):
+        domains = [start_url.allowed_domains for start_url in self.start_urls]
+        return list(reduce(set.union, domains, set()))
```

Comment:
For python 3 you would need to import reduce from functools.
You could also use:
```python
list(set(itertools.chain(*domains)))
```